modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2461,6 +2458,35 @@ private void _reindex(ObjectEntry objectEntry) throws PortalException {
 			objectDefinition.getClassName(), objectEntry.getObjectEntryId());
 	}
 
+	private void _setColumn(
+			int index, ObjectField objectField,
+			PreparedStatement preparedStatement,
+			Map<String, Serializable> values)
+		throws Exception {
+
+		Column<?, ?> column = dynamicObjectDefinitionTable.getColumn(
+			objectField.getDBColumnName());
+
+		if (StringUtil.equals(
+				objectField.getBusinessType(),
+				ObjectFieldConstants.BUSINESS_TYPE_MULTISELECT_PICKLIST)) {
+
+			String stringValue = String.valueOf(
+				values.get(objectField.getName()));
+
+			_setColumn(
+				preparedStatement, index, column.getSQLType(),
+				StringUtil.removeChars(
+					stringValue, CharPool.OPEN_BRACKET,
+					CharPool.CLOSE_BRACKET));
+		}
+		else {
+			_setColumn(
+				preparedStatement, index, column.getSQLType(),
+				values.get(objectField.getName()));
```

Comment:
@p-albuquerque can we move the `values.get(objectField.getName())` out as follows:

```
Column<?, ?> column = dynamicObjectDefinitionTable.getColumn(
    objectField.getDBColumnName());

Object value = values.get(objectField.getName());

if (StringUtil.equals(
        objectField.getBusinessType(),
        ObjectFieldConstants.BUSINESS_TYPE_MULTISELECT_PICKLIST)) {

    _setColumn(
        preparedStatement, index, column.getSQLType(),
        StringUtil.removeChars(
            String.valueOf(value), CharPool.OPEN_BRACKET,
            CharPool.CLOSE_BRACKET));
}
else {
    _setColumn(
        preparedStatement, index, column.getSQLType(), value);
}
```